### ISSUE TITLE
Changes in the rigidity transformation of the input point cloud cause network performance to degrade

### ISSUE THREAD

**Issue by cherwinsd** [hxwork/OMNet_Pytorch#10]
Hello, when I tested with your "os_rpmnet_clean" pre-trained model, I found that if "os_prnet_clean" is used as the transformation type of the input point cloud, the "mask" will mask all the points. What is the cause of this?

**Comment by hxwork** [comment]
Hi,

I haven't tested like this, but I think this is reasonable since PRNet and RPMNet apply different data pre-processing manners to generate partial-to-partial pairs. The global feature-based methods may be sensitive to these changes.

**Comment by cherwinsd** [comment]
Thank you for your reply. By the way, how many epochs have you trained in the "os_prnet_clean" model? After training for 200 epochs, I found that the value of "score" repeatedly fluctuated and did not decrease. And MSE (R) jumps between 10-14.

**Comment by cherwinsd** [comment]
I also noticed another problem while reading your code. In the "module. py" file, the "point_feat64" returned by the "OMNetEncoder" class is multiplied by the "mask". However, the local features returned in the network graph in your "readme. md" file are not multiplied by the "mask". Is the diagram or code incorrect?

**Comment by hxwork** [comment]
Hi,

For the first problem, please refer to the [params.json](https://github.com/hxwork/OMNet_Pytorch/blob/75dac389cae48e2d6579752a1dd7d3a8e68956e1/experiments/experiment_omnet/params.json#L30). For the second one, if I understand your question correctly, I think we indeed draw that multiplication in the figure of our pipeline. Please check it again.

**Comment by cherwinsd** [comment]
The following code is the "OMNetEncoder" class in the "[module. py](https://github.com/hxwork/OMNet_Pytorch/blob/main/model/module.py)" file.The bold part of the code is what I call "point_feat64", which is the local feature in your diagram.
`class OMNetEncoder(nn.Module):
    def __init__(self):
        super().__init__()
        self.conv_block1 = nn.Sequential(nn.Conv1d(3, 64, 1), nn.BatchNorm1d(64), nn.ReLU(inplace=True))

        self.conv_block2 = nn.Sequential(nn.Conv1d(64, 64, 1), nn.BatchNorm1d(64), nn.ReLU(inplace=True))

        self.conv_block3 = nn.Sequential(nn.Conv1d(64, 64, 1), nn.BatchNorm1d(64), nn.ReLU(inplace=True))

        self.conv_block4 = nn.Sequential(nn.Conv1d(64, 128, 1), nn.BatchNorm1d(128), nn.ReLU(inplace=True))

        self.conv_block5 = nn.Sequential(nn.Conv1d(128, 1024, 1), nn.BatchNorm1d(1024), nn.ReLU(inplace=True))

    def forward(self, x, mask=None):
        x = self.conv_block1(x)
        point_feat64 = self.conv_block2(x)
        x = self.conv_block3(point_feat64)
        x = self.conv_block4(x)
        point_feat1024 = self.conv_block5(x)

        if mask is None:
            L = [point_feat64, point_feat1024]
            glob_feat = torch.max(point_feat1024, dim=-1, keepdim=True)[0]
        else:
            L = [**point_feat64 * mask**, point_feat1024 * mask]
            glob_feat = torch.max(point_feat1024 * mask, dim=-1, keepdim=True)[0]

        return L, glob_feat`

**Comment by cherwinsd** [comment]
Hi,

The following code is the "OMNetEncoder" class in the "[module. py](https://github.com/hxwork/OMNet_Pytorch/blob/main/model/module.py)" file.The bold part of the code is what I call "point_feat64", which is the local feature in your diagram.

```
class OMNetEncoder(nn.Module):
    def __init__(self):
        super().__init__()
        self.conv_block1 = nn.Sequential(nn.Conv1d(3, 64, 1), nn.BatchNorm1d(64), nn.ReLU(inplace=True))

        self.conv_block2 = nn.Sequential(nn.Conv1d(64, 64, 1), nn.BatchNorm1d(64), nn.ReLU(inplace=True))

        self.conv_block3 = nn.Sequential(nn.Conv1d(64, 64, 1), nn.BatchNorm1d(64), nn.ReLU(inplace=True))

        self.conv_block4 = nn.Sequential(nn.Conv1d(64, 128, 1), nn.BatchNorm1d(128), nn.ReLU(inplace=True))

        self.conv_block5 = nn.Sequential(nn.Conv1d(128, 1024, 1), nn.BatchNorm1d(1024), nn.ReLU(inplace=True))

    def forward(self, x, mask=None):
        x = self.conv_block1(x)
        point_feat64 = self.conv_block2(x)
        x = self.conv_block3(point_feat64)
        x = self.conv_block4(x)
        point_feat1024 = self.conv_block5(x)

        if mask is None:
            L = [point_feat64, point_feat1024]
            glob_feat = torch.max(point_feat1024, dim=-1, keepdim=True)[0]
        else:
            L = [**point_feat64 * mask**, point_feat1024 * mask]
            glob_feat = torch.max(point_feat1024 * mask, dim=-1, keepdim=True)[0]

        return L, glob_feat
```

**Comment by hxwork** [comment]
Actually, it has no influence that whether `point_feat64` is multiplied by `mask` in the encoder since we use `Conv1d, kernel_size=1` and the concatenated feature will be multiplied by `mask`  in the feature fusion module ([ref link](https://github.com/hxwork/OMNet_Pytorch/blob/75dac389cae48e2d6579752a1dd7d3a8e68956e1/model/module.py#L58)).

**Comment by cherwinsd** [comment]
But this is different from the structure in your diagram. In the diagram, neither the "point_feat64" nor the input of "regression" module are multiplied by the "mask". And this has an impact. When I remove the masks in these two places, the network performance has changed.

**Comment by hxwork** [comment]
I think you misunderstood what I said. You should keep the mask multiplication which is applied to the input of the regression module. What I mean is that since the point-wise features will be multiplied by the mask before being fed into the regression module, whether the `point_feat64` is multiplied by the mask does not matters.

**Comment by cherwinsd** [comment]
I think I understand what you mean. Thank you for your help and patience
